etl/snapshot.py

```diff
@@ -220,6 +236,7 @@ def load_from_yaml(cls, filename: Union[str, Path]) -> "SnapshotMeta":
                     publication_date=str(publication_date) if publication_date else None,
                     publication_year=meta.pop("publication_year", None),
                 )
+            assert meta.get("origin") or meta.get("source")
```

Comment:
It'll actually raise an error in case it has neither origin nor source (it won't raise a warning if you have only source, but no origin). I'll change the error message.